test/jdk/java/awt/print/PrinterJob/PrintTextTest.java

```diff
@@ -21,193 +21,212 @@
-        ptt = new PrintTextTest(name, xfont, null, false);
-        p.add(ptt, BorderLayout.CENTER);
-        p.add(name, ptt);
-        book.append(ptt, portrait);
-        book.append(ptt, landscape);
+        name = "Page " + page++;
+        pt = new PrintText(name, xfont, null, false, preferredSize);
+        pane.add(pt, BorderLayout.CENTER);
+        pane.add(name, pt);
+        book.append(pt, portrait);
+        book.append(pt, landscape);
 
         Font yfont = font.deriveFont(AffineTransform.getScaleInstance(1, 1.5));
-        name = "Page " + new Integer(page++);
-        ptt = new PrintTextTest(name, yfont, null, false);
-        p.add(ptt, BorderLayout.CENTER);
-        p.add(name, ptt);
-        book.append(ptt, portrait);
-        book.append(ptt, landscape);
+        name = "Page " + page++;
+        pt = new PrintText(name, yfont, null, false, preferredSize);
+        pane.add(pt, BorderLayout.CENTER);
+        pane.add(name, pt);
+        book.append(pt, portrait);
+        book.append(pt, landscape);
 
         if (System.getProperty("os.name").startsWith("Windows")) {
             font = new Font("MS Gothic", Font.PLAIN, 12);
-            name = "Page " + new Integer(page++);
-            ptt = new PrintJAText(name, font, null, true);
-            p.add(ptt, BorderLayout.CENTER);
-            p.add(name, ptt);
-            book.append(ptt, portrait);
-            book.append(ptt, landscape);
+            name = "Page " + page++;
+            pt = new PrintJapaneseText(name, font, null, true, preferredSize);
+            pane.add(pt, BorderLayout.CENTER);
+            pane.add(name, pt);
+            book.append(pt, portrait);
+            book.append(pt, landscape);
 
             font = new Font("MS Gothic", Font.PLAIN, 12);
-            name = "Page " + new Integer(page++);
+            name = "Page " + page++;
             rotTx = AffineTransform.getRotateInstance(0.15);
-            ptt = new PrintJAText(name, font, rotTx, true);
-            p.add(ptt, BorderLayout.CENTER);
-            p.add(name, ptt);
-            book.append(ptt, portrait);
-            book.append(ptt, landscape);
+            pt = new PrintJapaneseText(name, font, rotTx, true, preferredSize);
+            pane.add(pt, BorderLayout.CENTER);
+            pane.add(name, pt);
+            book.append(pt, portrait);
+            book.append(pt, landscape);
         }
 
         pjob.setPageable(book);
 
-        JFrame f = new JFrame();
-        f.add(BorderLayout.CENTER, p);
-        f.addWindowListener(new WindowAdapter() {
-            public void windowClosing(WindowEvent e) {System.exit(0);}
+        JButton printButton = new JButton("Print");
+        printButton.addActionListener(event -> {
+            try {
+                if (pjob.printDialog()) {
+                    pjob.print();
+                }
+            } catch (PrinterException e) {
+                throw new RuntimeException(e.getMessage());
```

Comment:
```suggestion
                throw new RuntimeException(e.getMessage(), e);
```
The original exception could be very useful for debugging a failure.